### PR TITLE
Update the collation supported by TiDB

### PR DESCRIPTION
**Pull request by imp2002**
### What is changed, added or deleted? (Required)
Support the following sorting rules

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v5.1 (TiDB 5.1 versions)
- [x] v5.0 (TiDB 5.0 versions)
- [ ] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):
file:[character-set-and-collation.md](https://github.com/pingcap/docs-cn/blob/master/character-set-and-collation.md)
issue:https://github.com/pingcap/docs-cn/issues/5585

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by imp2002** [comment]
\label needs-cherry-pick-release-4.0
\label needs-cherry-pick-release-5.0

File: character-set-and-collation.md [hunk 1]
```diff
@@ -72,16 +72,21 @@ TiDB 支持以下排序规则：
...
+| ascii_bin                    | ascii   |   65 | Yes     | Yes      |       1 |
+| binary                       | binary  |   63 | Yes     | Yes      |       1 |
+| latin1_bin                   | latin1  |   47 | Yes     | Yes      |       1 |
+| utf8_bin                     | utf8    |   83 | Yes     | Yes      |       1 |
+| utf8_general_ci              | utf8    |   33 |         | Yes      |       1 |
+| utf8_unicode_ci              | utf8    |  192 |         | Yes      |       1 |
+| utf8mb4_bin                  | utf8mb4 |   46 | Yes     | Yes      |       1 |
+| utf8mb4_general_ci           | utf8mb4 |   45 |         | Yes      |       1 |
+| utf8mb4_unicode_ci           | utf8mb4 |  224 |         | Yes      |       1 |
+| utf8mb4_zh_pinyin_tidb_as_cs | utf8mb4 | 2048 |         | Yes      |       1 |
```
Comment (wjhuang2016) [inline character-set-and-collation.md:87]: utf8mb4_zh_pinyin_tidb_as_cs is not supported yet.

**Comment by TomShawn** [comment]
@imp2002 Any update?

**Comment by TomShawn** [comment]
@imp2002 To better manage PRs in this repo, I'll close this PR for now. If you have further update, you are welcome to reopen it.

**Pull request closed without merging**